tuist/tuist
# Add fixture generator for stress testing Tuist

Part of: https://github.com/tuist/tuist/issues/820 

### Short description 📝

To aid profiling Tuist large fixtures are needed.

### Solution 📦

Create a small command line tool to generate large fixtures

Usage:

```sh
swift run FixtureGenerator --projects 10 --targets 10 --sources 100
```

### Implementation 👩‍💻👨‍💻

- [x] Add fixture generator

### Test Plan 🛠

- Run `swift run` within `fixtures/stress_test`
- Run `tuist generate` within fixtures/stress_test/Fixture`
- Verify the generated fixure is valid
- Experiment with different number of sources / targets / projects to benchmark Tuist

### Questions

- Does this look like a reasonable approach to generating fixtures? 
- Any better tools / techniques we can apply here to make this easier to maintain / scale / extend?
